arrow/ipc/file_reader.go

```diff
@@ -388,34 +498,23 @@ func (src *ipcSource) buffer(i int) *memory.Buffer {
 		return memory.NewBufferBytes(nil)
 	}
 
-	raw := memory.NewResizableBuffer(src.mem)
+	var raw *memory.Buffer
 	if src.codec == nil {
-		raw.Resize(int(buf.Length()))
-		_, err := src.r.ReadAt(raw.Bytes(), buf.Offset())
-		if err != nil {
-			panic(err)
-		}
+		raw = memory.SliceBuffer(src.rawBytes, int(buf.Offset()), int(buf.Length()))
 	} else {
-		sr := io.NewSectionReader(src.r, buf.Offset(), buf.Length())
-		var uncompressedSize uint64
-
-		err := binary.Read(sr, binary.LittleEndian, &uncompressedSize)
-		if err != nil {
-			panic(err)
-		}
+		body := src.rawBytes.Bytes()[buf.Offset() : buf.Offset()+buf.Length()]
+		uncompressedSize := binary.LittleEndian.Uint64(body[:8])
 
-		var r io.Reader = sr
 		// check for an uncompressed buffer
 		if int64(uncompressedSize) != -1 {
```

Comment:
nit: while this is existing code, maybe it would be safer to have `uncompressedSize := int64(...)` so you don't have to remember to convert it on use and so it's consistent with above